### ISSUE TITLE
[Improvement] Support of Oracle as a datasource (Exploratory)

### ISSUE THREAD

**Issue by felixsuarez0727** [appsmithorg/appsmith#11274]
Supporting Oracle as a datasource has been getting some interest. While we want to prioritise this, finding a workaround is an option. We need to check if this is something worth investigating:

https://www.oracle.com/tools/technologies/faq-rest-data-services.html .

Or any other workaround that is easier to provide than to create a new Oracle datasource.

**Comment by felixsuarez0727** [comment]
**TL;DR:**

- As of now, there are ways to get data from an Oracle database that do not require a new DataSource (see below) . This option would require a setup and development from users. It is not hard to set up, but requires thoughtful design decisions.
    - Oracle REST Data Services
    - Dream Factory
    - OpenLegacy Oracle API
- Working on a plugin (like Twilio) could make it easier for our clients, but since the schema would vary from client to client (they are in control of the API) it would be better to stick to custom API calls.
- From my point of view, the potential of ORDS (and the others) lies as a way to meet customers' requirements while a final solution (DataSource) is being developed. No documentation was found on how to use this with Appsmith, blogs and videos could be created to track attention of Oracle users and understand their use cases and to get a better picture of what kind of priority this should be given.

## Oracle REST DATA Services + Appsmith API Calls

**Description**: ORDS is included with the Oracl SQL license. It can be used to easily create REST APIs from Oracle SQL databases.

**Pros:**

- Works on both relational or NoSQL Oracle DBs.
- Endpoints protected by privs/roles. OAUTH2 also available
- ORDS is included on the Oracle License. No additional 
  licenses are needed.

**Cons:**

- Requires ORDS setup on the side of the customer. Refer to 
  [AutoREST][AutoREST] feature for an easy solution to this.

## Oracle REST DATA Services + Appsmith Plugin

**Description:** An Appsmith plugin could make it easier for a customer to get data from an Oracle database once ORDS has been implemented. The feasability of this option requires testing due to how variable the APIs could be.

**Pros:**

- Works on both relational or NoSQL Oracle DBs.
- Endpoints protected by privs/roles. OAUTH2 also available
- ORDS is included on the Oracle License. No additional licenses are needed.

**Cons:**

- Requires ORDS setup on the side of the customer. Refer to [AutoREST][AutoREST] feature for an easy solution to this.

## Oracle DB + DreamFactory + Appsmith API Calls

**Description:** DreamFactory is a 3rd party service that creates REST APIs from SQL backends (Oracle included). First impressions indicate that it's easier to setup than ORDS.

**Pros:**

- Simplifies the creation of an API based on your SQL database (Oracle included)

**Cons:**

- Requires usage of a 3rd party platform/app. 

## Oracle DB + OpenLegacy Oracle API + Appsmith API Calls

**Description:** Another 3rd party service that creates APIs from SQL backends. Feature matching with DreamFactory requires research.

**Pros:**

- Simplifies the creation of an API based on your Oracle SQL database.

**Cons:**

- Requires usage of a 3rd party platform/app.

**Comment by rohan-arthur** [comment]
How-tos
ORDS AutoREST: https://oracle-base.com/articles/misc/oracle-rest-data-services-ords-autorest 
Dreamfactory: https://guide.dreamfactory.com/docs/generating-a-database-backed-api/#generating-a-mysql-backed-api
Openlegacy: https://www.openlegacy.com/blog/oracle-api
